### İş Uygulaması: Web Sitesinde Geçirilen Süre

*Elimizde bir grup kullanıcının web sitesinde geçirdikleri süreler ile ilgili veri var. CEO Ortalama geçirilen sürenin 170 sn olduğu iddia ediyor. Fakat bir Veri Bilimci bu veriye çok güvenemiyor ve bu verilerin hipotez testine tutulması gerektiğini savunuyor.*

- **Problem:**
  Web sitemizde geçirilen ortalama süre gerçekten 170 saniye mi?

- **Detaylar:**
  - Yazılımlardan elde edilen web sitesinde geçirilen ort. süreler var.
  - Bu veriler incelendiğinde bir yönetici ya da çalışanımız bu değerlerin böyle olmadığına yönelik düşünceler taşıyor ve bu durumu test etmek istiyorlar.


$$
H_0: \mu = 170
$$
$$
H_1: \mu \neq 170
$$

**Test istatistiği:**

$$
t = \frac{\bar{x} - \mu_0}{\frac{s}{\sqrt{n}}}
$$


In [1]:
# Web sitesinde zaman geçiren kullanıcıların geçirdikleri süreler
import numpy as np

olcumler = np.array([
    17, 160, 234, 149, 145, 107, 197, 75, 201, 225, 211, 119,
    157, 145, 127, 244, 163, 114, 145, 65, 112, 185, 202, 146,
    203, 224, 203, 114, 188, 156, 187, 154, 177, 95, 165, 50, 110,
    216, 138, 151, 166, 135, 155, 84, 251, 173, 131, 207, 121, 120
])

In [2]:
olcumler[0 : 10]

array([ 17, 160, 234, 149, 145, 107, 197,  75, 201, 225])

In [3]:
import scipy.stats as stats 
stats.describe(olcumler)

DescribeResult(nobs=np.int64(50), minmax=(np.int64(17), np.int64(251)), mean=np.float64(154.38), variance=np.float64(2578.0363265306123), skewness=np.float64(-0.32398897278694483), kurtosis=np.float64(-0.05849823498415985))

*Ölçüm ortalamasının 154.38 olduğunu görüyoruz. İddia edilen ortalamadan düşük,yani burada ölçüm yaparak kesin bir şekilde iddianın yanlış olduğunu varsayabiliriz fakat bu değerin şansa bağlı olarak çıkıp çıkmadığına nasıl emin olabiliriz? Yukarıda örnek 50 ölçüm aldık,başka bir 50 ölçüm değerinin ortalamasının 170'den yukarı olmayacağına nasıl emin olabiliriz ? İşte kesin olarak yapılan bütün ölçüm değerlerinin ortalamasının 170'den küçük olduğunu ispatlayabilmek için(burada da tek bir örneklem olduğu için) T testine başvuruyoruz. Bütün hipotez testlerinin amacı zaten şansı, insan kararlarını ortadan kaldırmak,elimizdeki ölçümleri bilimsel bir dayanağa oturtmak ve ona göre sağlıklı kararlar almaktır.*

*Hipotez testi der ki ben sana bu bilimsel dayanağı sağlarım.Ama birtakım ön şartlarım,varsayımlarım var:*

In [4]:
#Varsayımlar
#Normallik Varsayımı
#Shapiro - Wilks Testi

*Elimizdeki ölçümlerinin normal dağılım olup olmadığını ölçmek için yapılan testtir.*

*H0: Örnek dağılımı ile teorik normal dağılımı arasında istatistiksel olarak anlamlı bir fark yoktur.*<br>
*H1: Örnek dağılımı ile teorik normal dağılımı arasında istatistiksel olarak fark vardır.*<br>
*Not: Buradaki H0 ve H1 Shapiro-Wilks testinin hipotezleri,asıl hipotezlerle karıştırılmamalı.*

In [5]:
from scipy.stats import shapiro
print("T Hesap İstatistiği: " + str(shapiro(olcumler)[0]))
print("p-value değeri: " + str(shapiro(olcumler)[1]))

T Hesap İstatistiği: 0.9853103897061022
p-value değeri: 0.7848671719373956


*p-value değerimiz alpha değerinden küçük değil.Yani H0 reddedilemez.İstediğimiz de buydu.Normallik varsayımımız sağlanıyor.Yani Tek Örneklem T testi uygulanabilir.*

In [6]:
#Hipotez Testinin Uygulanması

In [7]:
stats.ttest_1samp(olcumler, popmean = 170)

TtestResult(statistic=np.float64(-2.1753117985877966), pvalue=np.float64(0.034460415195071474), df=np.int64(49))

*p-value değerimiz alpha değerinden küçük. H0'ı reddektik. Yani web sitesinde geçirilen ortalama süre 170 sn'den farklıdır.Ölçüm ortalamasından da bunu gördük.*

### ÖZET

*Shapiro - Wilks testi, Tek Örneklem T Testi öncesinde elimizdeki örnek dağılımının normal dağılımı ile uygunluğunu test eder.Buna neden gerek duyarız ? Çünkü T Örneklem Testi örnek dağılımının normal dağılım ile uyumlu olduğunu varsayar. Eğer gerçekten elimizdeki önrek dağılımı normal dağılımı ile uygun değilse bu varsayım yanlıştır ve T Testinin sonuçları güvenilir olmayabilir. Bu varsayım kontrolünü yaparak daha sonra yapacağımız hipotez testinin sonuçlarının güvenilir olduğunu doğruluyoruz.* 